acceptable-albatrosses/albatrosses_hub/authentication/tests.py

```diff
@@ -0,0 +1,152 @@
+        found = resolve(reverse("authentication:register"))
+        self.assertEqual(found.func, register_page)
+
+    def test_login_using_login_func(self):
+        found = resolve(reverse("authentication:login"))
+        self.assertEqual(found.func, login_page)
+
+    def test_logout_using_logout_func(self):
+        found = resolve(reverse("authentication:logout"))
+        self.assertEqual(found.func, logout_page)
+
+
+class AuthViewTest(TestCase):
+    """Class to perform some tests related to views."""
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(username="tester", email="test@example.com", password="tester12345")
+
+    def test_logged_in_user_get_redirect_on_register_page(self):
+        request = self.factory.get(reverse("authentication:register"))
+        request.user = self.user
+
+        response = register_page(request)
+        self.assertRedirects(response, "/", fetch_redirect_response=False)
+
+    def test_registered_email_can_not_be_registered(self):
+        request = self.factory.post(reverse("authentication:register"))
+        request.user = AnonymousUser()
+        request.POST = request.POST.copy()
+        request.POST["email"] = "test@example.com"
+
+        response = register_page(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_registered_username_can_not_be_taken(self):
+        request = self.factory.post(reverse("authentication:register"))
+        request.user = AnonymousUser()
+        request.POST = request.POST.copy()
+        request.POST["email"] = "test2@example.com"
+        request.POST["username"] = "tester"
+
+        response = register_page(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_password_is_different_with_retyped_password(self):
+        request = self.factory.post(reverse("authentication:register"))
+        request.user = AnonymousUser()
+        request.POST = request.POST.copy()
+        request.POST["email"] = "test2@example.com"
+        request.POST["username"] = "tester2"
+        request.POST["password"] = "tester2_12345"
+        request.POST["re_password"] = "tester_12345"
+
+        response = register_page(request)
+        self.assertEqual(response.status_code, 200)
+
+    def test_register_new_user(self):
+        request = self.factory.post(reverse("authentication:register"))
+        request.user = AnonymousUser()
+        request.POST = request.POST.copy()
+        request.POST["email"] = "test2@example.com"
+        request.POST["username"] = "tester2"
+        request.POST["password"] = "tester2_12345"
+        request.POST["re_password"] = "tester2_12345"
+
+        response = register_page(request)
+        self.assertRedirects(response, reverse("authentication:login"), fetch_redirect_response=False)
+
+    def test_logged_in_user_get_redirect_on_login_page(self):
+        request = self.factory.get(reverse("authentication:login"))
```

Comment:
This is such a method that could use the `django.test.Client.login` method instead of the regular authentication flow. If your authentication flow is broken by a bad update, you could still test the redirection for logged in users by using the `Client` method to log them in.